Lib/_pyrepl/reader.py

```diff
@@ -701,3 +704,10 @@ def bind(self, spec: KeySpec, command: CommandName) -> None:
     def get_unicode(self) -> str:
         """Return the current buffer as a unicode string."""
         return "".join(self.buffer)
+
+    def call_PyOS_InputHook(self):
+        if self.input_hook_addr is None:
+            self.input_hook_addr = ctypes.c_void_p.in_dll(ctypes.pythonapi, 'PyOS_InputHook').value
```

Comment:
I don't think we can cache this. The value can change at any point and we should always pick up the current pointer value.

Reply:
What I intent to cache is the address of the symbol not the content of the pointer, will add a test for this and fix it if I cached the wrong thing